loudsmilestudios/TetraForce
# Improve Versioning System & Disconnect Handling

![image](https://user-images.githubusercontent.com/21203171/127404156-3a97293c-d92e-430f-b397-8c890fdb727e.png)

### Summary
All builds now generate a version that is validated when connecting to a server. This all has disconnection handling. Previously, when a server shutdown while a player was connected the client would freeze. Now, a server disconnecting will send connected players back to the lobby.

This PR adds:
- Game Version
  - You can reference this anywhere in the project with `global.version`
- Server Kicking
  - There is a function called `kick_player` in `network.gd` that can be used by the network master to easily kicked people from the game
- Server Disconnection handling

### Testing
1. Download a versioned build: https://github.com/loudsmilestudios/TetraForce/actions/runs/1076559012
2. Checkout branch
3. Try connecting to build from branch and vice versa
  - Only your Godot build should be able to connect to itself and the version build to other versions of itself

***NOTE:** This will cause all clients v0.1.0 and earlier to freeze when trying to connect to a new server because it does not have any disconnection handling/*

[**Has this been tested in multiplayer?**](https://github.com/loudsmilestudios/TetraForce/wiki/How-to-test-multiplayer) yes
